jena-fuseki-kafka-module/src/main/java/org/apache/jena/fuseki/kafka/FKBatchProcessor.java

```diff
@@ -161,12 +214,12 @@ private Timer batchStart(String topic, long lastOffsetState, int count, long num
         return timer;
     }
 
-    private long batchProcess(String topic, ConsumerRecords<String, RequestFK> cRecords) {
+    private long batchProcess(String topic, List<ConsumerRecords<String,RequestFK>> recordList) {
         if ( transactional == null ) {
             // No transactional set. Assume the fkProcessor.process knows what it is doing.
-            return execBatch(cRecords);
+            return execBatch(recordList);
         }
-        return transactional.calculate(()->execBatch(cRecords));
+        return transactional.calculate(()->execBatch(recordList));
```

Comment:
Since we're cleaning up anyway looking at the Javadoc should this perhaps be `calculateWrite()` rather than plain `calculate()`?

We're always expecting to write to the database here and `calculate()` suggests that the transaction could be aborted if another write transaction happens in the meantime?

Reply:
Yeah, I was trying not to pull at that thread.... start fixing one thing, then another ... then another. I'm tempted to update the processBatch() method to pass in the payload and count to save from re-calculating them. 

I'll make the change and run the tests - should be fine I think.